components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -850,26 +850,35 @@ private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpSe
 
         samlssoTokenIdCookie.setSecure(true);
         samlssoTokenIdCookie.setHttpOnly(true);
+        samlssoTokenIdCookie.setPath("/");
+        samlssoTokenIdCookie.setMaxAge(defaultMaxAge);
+
         if (samlssoTokenIdCookieConfig != null) {
-            samlssoTokenIdCookie.setMaxAge(samlssoTokenIdCookieConfig.getMaxAge() > 0 ?
-                    samlssoTokenIdCookieConfig.getMaxAge() :
-                    defaultMaxAge);
-            samlssoTokenIdCookie.setDomain(samlssoTokenIdCookieConfig.getDomain());
-        } else {
-            samlssoTokenIdCookie.setMaxAge(defaultMaxAge);
+            int age = samlssoTokenIdCookieConfig.getMaxAge() > 0 ? samlssoTokenIdCookieConfig.getMaxAge() :
+                    defaultMaxAge;
```

Comment:
In this case, we give priority to max age value configured through identity.xml over session idle time out configured for that tenant. Is this correct?